docs/testing.md

```diff
@@ -43,7 +49,51 @@ Or if you want to run bats from a different repository or branch, edit `centos7-
         bats_forklift_repo: https://github.com/<YOUR_NAME>/forklift.git
         bats_forklift_version: your-branch
 
-## Pipeline Testing
+### To run Bats on an existing VM for development
+
+1. Clone forklift onto your VM (it's recommended to clone into a directory named other than `forklift` so you don't get mixed up):
+```
+git clone git@github.com:theforeman/forklift.git forklift-devel
+```
+2. Clone the Bats repo:
+```
+git clone https://github.com/sstephenson/bats.git
+```
+3. Install Bats on your VM in the place where `katello-bats` expects it:
+```
+cd bats
+sudo ./install.sh /opt/bats
+```
+4. Make sure `bats` is in your `PATH`:
+```
+export PATH="/opt/bats/bin:$PATH"
+```
+5. Make sure the `bats` command works:
+```
+bats
+Bats 0.4.0
+Usage: bats [-c] [-p | -t] <test> [<test> ...]
+```
+6. Instead of using forklift's `bats/install.sh` to copy our Bats tests, create symlinks instead:
+```
+cd <forklift_devel_directory>/bats
+sudo ln -s -v -t /opt/bats $(pwd)/*.bats
+sudo ln -s -v -t /opt/bats $(pwd)/*.bash
+sudo mkdir -p /opt/bats/fixtures
+sudo ln -s -v -t /opt/bats/fixtures $(pwd)/fixtures/*
+```
+(Replace `<forklift_devel_directory>` with the directory to which you cloned forklift)
+8. Make sure you have a working `hammer` command.
+9. Make sure your Foreman server is running
+10. Run the test suite:
+```
+cd <forklift_devel_directory>/bats
+sudo ./katello-bats <devel|content|proxy|virt-whom>
```

Comment:
Is this whole process meant to be done as root? I was using my vagrant user and am getting this error:
```
[vagrant@centos7-katello-devel-stable bats{master}]$ sudo ./katello-bats devel
./katello-bats: line 55: bats: command not found
```

Reply:
I had some problems with that too, I think I had to add the bats binary to $PATH.

Reply:
@jeremylenz I assume most people running this would run into the same thing, should we suggest this whole process should be done as root?